scripts/tags_and_languages.py

```diff
@@ -0,0 +1,338 @@
+            if tag == old_tag_with_lc:
+                tags_list[i] = new_tag
+                updated = True
+                break
+    if not updated:
+        for i, tag in enumerate(tags_list):
+            if tag == old_tag:
+                tags_list[i] = new_tag
+                updated = True
+                break
+
+
+    updated_tags_text = ", ".join(tags_list)
+
+    return updated_tags_text, updated
+
+
+time_start = datetime.now()
+
+for tag_type in tag_types:
+    print(f"retrieve_taxonomy_tag from static url - time {datetime.now()-time_start}")
+    retrieve_taxonomy_tag(tag_type)
+
+
+    print(f"call products from hf - time {datetime.now()-time_start}")
+    tags_to_update_df = conn.execute(f'''
+    WITH products_tags AS (
+        SELECT
+            hf_dataset.code,
+            hf_dataset.lang,
+            hf_dataset.{tag_type},
+            unnest(hf_dataset.{tag_type}_tags) AS {tag_type}_lc_and_tag,
+            str_split(unnest(hf_dataset.{tag_type}_tags), ':')[2] AS {tag_type}_tag
+        FROM 'hf://datasets/openfoodfacts/product-database/food.parquet' hf_dataset
+        WHERE {tag_type}_tags IS NOT NULL
+    ),
+    products_unknown_tags AS (
+        SELECT
+            products_tags.code,
+            products_tags.lang,
+            products_tags.{tag_type},
+            products_tags.{tag_type}_lc_and_tag,
+            products_tags.{tag_type}_tag,
+            taxonomy_tags.tag,
+            taxonomy_tags.lc,
+            taxonomy_tags.lc_tag_name,
+            taxonomy_tags.lc_tag_name_concat
+        FROM products_tags
+        LEFT JOIN {tag_type}_tags taxonomy_tags
+        ON products_tags.{tag_type}_lc_and_tag = taxonomy_tags.tag
+        WHERE taxonomy_tags.lc_tag_name IS NULL
+    )
+    SELECT
+        products_unknown_tags.code AS products_code,
+        products_unknown_tags.lang AS products_lang,
+        products_unknown_tags.{tag_type} AS products_{tag_type},
+        products_unknown_tags.{tag_type}_lc_and_tag AS products_{tag_type}_unknown_tag,
+        products_unknown_tags.{tag_type}_tag AS products_{tag_type}_unknown_tag_name,
+        taxonomy_tags.tag AS taxonomy_tag_id,
+        taxonomy_tags.lc AS taxonomy_tag_lc,
+        taxonomy_tags.lc_tag_name AS taxonomy_tag_name,
+        taxonomy_tags.lc_tag_name_concat AS taxonomy_tag
+    FROM products_unknown_tags
+    LEFT JOIN {tag_type}_tags taxonomy_tags
+    ON products_unknown_tags.{tag_type}_tag = taxonomy_tags.lc_tag_name
+    WHERE taxonomy_tags.lc_tag_name IS NOT NULL
+    AND taxonomy_tags.lc != 'xx'
+    AND products_unknown_tags.{tag_type}_lc_and_tag != taxonomy_tags.lc_tag_name_concat
+    ''').df()
+    # remark:
+    #   AND products_unknown_tags.{tag_type}_lc_and_tag != taxonomy_tags.lc_tag_name_concat -> to prevent fr:angleterre and fr:angleterre
+
+    print(f"prepare log table - time {datetime.now()-time_start}")
+    conn_db = duckdb.connect(database='tags_and_languages.db', read_only=False)
```

Comment:
maybe use a different name from above, it's more fail safe !

eg: `updated_db`